File: levelledmobs-plugin-bukkit/src/main/java/io/github/arcaneplugins/levelledmobs/bukkit/integration/type/nbt/NbtProvider.java
package io.github.arcaneplugins.levelledmobs.bukkit.integration.type.nbt;

import org.bukkit.entity.LivingEntity;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public interface NbtProvider {

    @NotNull
    NbtModificationResult addNbtTag(final ItemStack itemStack, final String tag);

    @NotNull
    NbtModificationResult addNbtTag(final LivingEntity entity, final String tag);

    @NotNull
    String getNbtDump(final ItemStack itemStack);

    @NotNull
    String getNbtDump(final LivingEntity entity);

}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/bukkit/integration/type/nbt/NbtProvider.kt
package io.github.arcaneplugins.levelledmobs.bukkit.integration.type.nbt

import org.bukkit.entity.LivingEntity
import org.bukkit.inventory.ItemStack

interface NbtProvider {
    fun addNbtTag(itemStack: ItemStack, tag: String): NbtModificationResult

    fun addNbtTag(entity: LivingEntity, tag: String): NbtModificationResult

    fun getNbtDump(itemStack: ItemStack): String

    fun getNbtDump(entity: LivingEntity): String
}